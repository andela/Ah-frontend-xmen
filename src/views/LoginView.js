import React from 'react';
import Login from '../components/Login';
import SocialAuthView from './socialAuthView/socialAuthView';

const LoginView = props => (
  <div>
    <SocialAuthView />
    <Login />

  </div>
);

export default LoginView;
