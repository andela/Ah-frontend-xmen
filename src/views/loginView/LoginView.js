import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Login from '../../components/Login/loginForm';
import loginAction from '../../actions/loginActions/loginAction';
import SocialAuthView from '../socialAuthView/socialAuthView';

const jwt = require('jsonwebtoken');

export class LoginView extends Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.token) {
      localStorage.setItem('token', nextProps.token);
      const decoded = jwt.decode(nextProps.token);
      if (decoded) {
        const username = decoded.user_data.split(' ')[1];
        localStorage.setItem('username', username);
      }
      setTimeout(() => window.location.reload('/'), 100);
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { loginAction } = this.props;
    const credentials = {
      email: event.target.elements.email.value,
      password: event.target.elements.password.value,
    };
    loginAction(credentials);
  };

  openSignup = (e) => {
    this.props.onOpenSignup(e);
  }

  render() {
    const { errors } = this.props;
    const { isSuccessful } = this.props;
    return (
      <div>
        <h3 className="text-center text-primary mt-4">Sign in with</h3>
        <SocialAuthView />
        <Login
          onSubmit={this.handleSubmit}
          errors={errors}
          isSuccessful={isSuccessful}
          onOpenSignup={this.openSignup}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  token: state.loginReducer.token,
  errors: state.loginReducer.errors,
  isSuccessful: state.loginReducer.isSuccessful,
});

LoginView.propTypes = {
  token: PropTypes.string,
  errors: PropTypes.shape({
    email: PropTypes.any,
    password: PropTypes.any,
    error: PropTypes.any,
  }),

};

LoginView.defaultProps = {
  token: '',
  errors: {},
};

export default connect(mapStateToProps, { loginAction })(LoginView);
