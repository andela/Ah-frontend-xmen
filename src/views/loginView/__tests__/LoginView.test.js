import React from 'react';
import { shallow } from 'enzyme';
import { LoginView } from '../LoginView';

describe('LoginView', () => {
  it('should render without crashing', () => {
    const props = {};
    const instance = new LoginView(props);
    const wrapper = shallow(
      instance.render(),
    );
    expect(wrapper).toMatchSnapshot();
  });


  it('should reload on new token', () => {
    const props = {
      token: '',
    };
    global.setTimeout = jest.fn();

    const wrapper = shallow(<LoginView {...props} />);
    wrapper.instance().componentWillReceiveProps({
      token: '123',
    });
    expect(global.setTimeout).toHaveBeenCalled();
  });

  it('should not reload if there is no token received', () => {
    const props = {
      token: '',
    };
    global.setTimeout = jest.fn();

    const wrapper = shallow(<LoginView {...props} />);
    wrapper.instance().componentWillReceiveProps({});
    expect(global.setTimeout).not.toHaveBeenCalled();
  });

  it('should call the login action', () => {
    const props = { loginAction: jest.fn() };
    const wrapper = shallow(<LoginView {...props} />);

    const event = {
      preventDefault: jest.fn(),
      target:
      {
        elements: {
          email: {
            value: 'testemail@mail.com',
          },
          password: {
            value: 'Password1',
          },
        },
      },
    };
    wrapper.instance().handleSubmit(event);
    expect(wrapper.instance().props.loginAction).toHaveBeenCalled();
  });
});
