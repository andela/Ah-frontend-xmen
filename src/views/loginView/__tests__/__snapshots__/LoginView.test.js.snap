// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView should render without crashing 1`] = `ShallowWrapper {}`;
